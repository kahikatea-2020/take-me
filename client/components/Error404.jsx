import React from 'react'
import { Link } from 'react-router-dom'
import { Button } from 'semantic-ui-react'

class Error404 extends React.Component {
  render () {
    return (
      <>
        <div className="ui placeholder segment">
          <div className="ui icon header">
            <i className="search icon"></i>
            <h1 style={{ color: 'red', fontSize: '40px' }}> 404 </h1>
            <h2>Sorry, we can't find the page you requested</h2>
            <h3>You may have clicked on a broken link or entered an incorrect URL</h3>
          </div>
          <br />
          <div className="inline">
            <Button id='home' as={Link} to='/' >Home</Button>
            <Button id='home' as='a' onClick={() => this.props.history.goBack()} to='/' >Back</Button>
          </div>
        </div>
      </>
    )
  }
}

export default Error404
