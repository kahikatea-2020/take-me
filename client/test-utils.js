import React from 'react'
import { Provider } from 'react-redux'
import { createStore, applyMiddleware, compose } from 'redux'
import { MemoryRouter as Router } from 'react-router-dom'
import thunkMiddleware from 'redux-thunk'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import reducer from './reducers'
const enhancer = compose(applyMiddleware(thunkMiddleware))

export const renderWithRedux = (
  ui,
  { initialState, initialEntries = ['/'], store = createStore(reducer, initialState, enhancer) } = {}
) => {
  return {
    ...render(
      <Provider store={store}>
        <Router initialEntries={initialEntries} initialIndex={0}>
          {ui}
        </Router>
      </Provider>
    ),
    store
  }
}
