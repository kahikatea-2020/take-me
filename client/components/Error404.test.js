import React from 'react'
import { renderWithRedux } from '../test-utils'
import '@testing-library/jest-dom/extend-expect'

import Error404 from './Error404'

test('Test Error 404 component works', () => {
  const rendered = renderWithRedux(<Error404 />)
  expect(rendered).toMatchSnapshot()
})
