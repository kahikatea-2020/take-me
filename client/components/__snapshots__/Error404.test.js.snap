// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Error 404 component works 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ui placeholder segment"
      >
        <div
          class="ui icon header"
        >
          <i
            class="search icon"
          />
          <h1
            style="color: red; font-size: 40px;"
          >
             404 
          </h1>
          <h2>
            Sorry, we can't find the page you requested
          </h2>
          <h3>
            You may have clicked on a broken link or entered an incorrect URL
          </h3>
        </div>
        <br />
        <div
          class="inline"
        >
          <a
            class="ui button"
            href="/"
            id="home"
            role="button"
          >
            Home
          </a>
          <a
            class="ui button"
            id="home"
            role="button"
            to="/"
          >
            Back
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ui placeholder segment"
    >
      <div
        class="ui icon header"
      >
        <i
          class="search icon"
        />
        <h1
          style="color: red; font-size: 40px;"
        >
           404 
        </h1>
        <h2>
          Sorry, we can't find the page you requested
        </h2>
        <h3>
          You may have clicked on a broken link or entered an incorrect URL
        </h3>
      </div>
      <br />
      <div
        class="inline"
      >
        <a
          class="ui button"
          href="/"
          id="home"
          role="button"
        >
          Home
        </a>
        <a
          class="ui button"
          id="home"
          role="button"
          to="/"
        >
          Back
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
    Symbol(observable): [Function],
  },
  "unmount": [Function],
}
`;
